Obsidian.API/Plugins/PluginBase.cs

```diff
@@ -20,6 +20,18 @@ public abstract class PluginBase
 
         private Type typeCache;
 
+        public PluginBase(Type TypeCache)
+        {
+            typeCache = TypeCache;
+        }
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+        public PluginBase()
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+        {
+
+        }
+
```

Comment:
The best approach to take here would've been writing a method that handles the null-checking where it sets `typeCache` to `GetType()` when `null` and returns the result, replacing most of the `typeCache ??= GetType()` calls. After that it would be safe to make `typeCache` nullable.

Reply:
Have altered the empty constructor to `typeCache ??= GetType()` this now means the other one is not needed.